domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -12,24 +13,27 @@ namespace API.Controllers;
 [ApiController]
 public class TransferAgreementsController : ControllerBase
 {
-    private readonly ITransferAgreementService transferAgreementService;
+    private readonly ITransferAgreementRepository transferAgreementRepository;
 
-    public TransferAgreementsController(ITransferAgreementService transferAgreementService) => this.transferAgreementService = transferAgreementService;
+    public TransferAgreementsController(ITransferAgreementRepository transferAgreementRepository) => this.transferAgreementRepository = transferAgreementRepository;
 
     [HttpPost("api/transfer-agreements")]
     public async Task<ActionResult> Create([FromBody] CreateTransferAgreement request)
     {
+        var actor = User.FindActorClaim();
+        var subject = User.FindSubjectClaim();
+
         var transferAgreement = new TransferAgreement
         {
-            SenderId = Guid.Parse(User.FindFirstValue("sub")),
-            ActorId = User.FindFirstValue("atr"),
-            StartDate = request.StartDate.UtcDateTime,
-            EndDate = request.EndDate.UtcDateTime,
+            StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
+            EndDate = DateTimeOffset.FromUnixTimeSeconds(request.EndDate),
+            ActorId = actor,
+            SenderId = Guid.Parse(subject),
             ReceiverTin = request.ReceiverTin
         };
 
-        var result = await transferAgreementService.CreateTransferAgreement(transferAgreement);
+        var result = await transferAgreementRepository.CreateTransferAgreement(transferAgreement);
 
-        return Ok(result);
+        return Created($"api/transfer-agreements/{result.Id}", result);
```

Comment:
An object of type `TransferAgreement` is returned, which is a domain object/entity/internal data model (I am not sure what you call it). Should it not be mapped to an API model?

Reply:
It could. But the internal model also acts as the layout of the response. One could argue that creating a separate model, which represents the resonse (Something like "CreateTransferAgreementResponse"), would establish separation of concerns, more "correctly". I am leaving it as is for now, as i will be creating new response DTO's, when implementing show and list.